browse/services/document/metadata.py

```diff
@@ -460,38 +478,62 @@ def parse_abs_file(filename: str) -> DocMetadata:
             raise AbsParsingException('Could not extract submitter data.')
         name = from_match.group('name').rstrip()
         email = from_match.group('email') or None
-        fields['submitter'] = Submitter(name=name, email=email)
 
         # get the version history for this particular version of the document
         if not len(parsed_version_entries) >= 1:
             raise AbsParsingException('At least one version entry expected.')
 
-        AbsMetaSession._parse_version_entries(
-            fields=fields,
-            version_entry_list=parsed_version_entries
+        (version, version_history, arxiv_id_v) = AbsMetaSession._parse_version_entries(
+            arxiv_id=arxiv_id, version_entry_list=parsed_version_entries
         )
+
+        # TODO type ignore: possibly mypy #3937, also see #5389
+        arxiv_identifier=Identifier(arxiv_id=arxiv_id)
+
         # named (key-value) fields
-        AbsMetaSession._parse_metadata_fields(fields=fields,
-                                              key_value_block=misc_fields)
-        if 'categories' not in fields and fields['arxiv_identifier'].is_old_id:
-            fields['categories'] = fields['arxiv_identifier'].archive
+        if 'categories' not in fields and arxiv_identifier.is_old_id:
+            fields['categories'] = arxiv_identifier.archive
 
         if not all(rf in fields for rf in REQUIRED_FIELDS):
             raise AbsParsingException(f'missing required field(s)')
 
         # some transformations
         categories = fields['categories'].split()
-        fields['primary_category'] = Category(id=categories[0])
-        fields['secondary_categories'] = [
-            Category(id=x) for x in categories[1:] if len(categories) > 1
-        ]
-        if 'license' in fields:
-            fields['license'] = License(recorded_uri=fields['license'])
 
-        return DocMetadata(**fields)
+        doc_license: License = \
+            License() if 'license' not in fields else License(recorded_uri=fields['license'])  # type: ignore
+
+        return DocMetadata(  # type: ignore
```

Comment:
While this is clearer than `return DocMetadata(**fields)`, it's problematic. All the fields that have been commented out with question marks (`# journal_ref=???`) are no longer included.

Reply:
@bbarker this PR was merged before I had a chance to review it, but I wanted to draw your attention to this comment.

Reply:
Sorry about that - I'm out today, but will add this back in asap, possibly today. I'll attempt 1) add back all the fields without them being optional, 2) make them optional if it seems they are missing sometimes, 3) leave it as is otherwise, as I assume that means the fields don't exist.

Reply:
I guess this will also be a good chance to add some tests around these fields to check they are indeed added?

Reply:
No worries. I should have made the tests more comprehensive when I originally wrote `test_abs_parser.py`.  None of those examples actually have non-`None` values for those fields.

Reply:
@mhl10 working for fixes in #22